pycobertura/reporters.py

```diff
@@ -1,90 +1,64 @@
-from collections import namedtuple
 from jinja2 import Environment, PackageLoader
 from pycobertura.cobertura import CoberturaDiff
-from pycobertura.utils import green, rangify, red
+from pycobertura.utils import green, red, stringify
 from pycobertura.templates import filters
 from tabulate import tabulate
 
 
 env = Environment(loader=PackageLoader("pycobertura", "templates"))
 env.filters["line_status"] = filters.line_status
 env.filters["line_reason"] = filters.line_reason_icon
+env.filters["is_not_equal_to_dash"] = filters.is_not_equal_to_dash
+env.filters["misses_color"] = filters.misses_color
 
-row_attributes = ["filename", "total_statements", "total_misses", "line_rate"]
-file_row = namedtuple("FileRow", row_attributes)
-file_row_missed = namedtuple("FileRowMissed", row_attributes + ["missed_lines"])
+headers_without_missing = ["Filename", "Stmts", "Miss", "Cover"]
+headers_with_missing = ["Filename", "Stmts", "Miss", "Cover", "Missing"]
 
 
 class Reporter(object):
     def __init__(self, cobertura):
         self.cobertura = cobertura
+        self.files_info = self.cobertura.files()
```

Comment:
Just so you know, `self.cobertura.files()` is already cached/memoized, so calling it multiple times shouldn't incur additional XML parsing. Saying that in case you were using `self.files_info` as a cache.

Reply:
Yes, had to with your comment here: https://github.com/aconrad/pycobertura/pull/124#discussion_r784635006

Reply:
But I think we can still keep it or would you prefer to have `self.cobertura.files()` explicitly everywhere?

Reply:
I'd prefer using `self.cobertura.files()`.

I see how my comment might've been confusing. When I talked about performance, I meant the overhead of the for-loop in Python since we're doing it a bunch of times now (vs. once previously). It didn't mean the cost of computing `self.cobertura.files()` (as I said above, this call is computed once and then memoized).

Don't worry about the performance part, I think it's negligible. Let's use `self.cobertura.files()` and we'll be good to merge this PR! 🎉

Reply:
done